scripts/setup-ubuntu.sh

```diff
@@ -39,6 +39,20 @@ DEPENDENCY_DIR=${DEPENDENCY_DIR:-$(pwd)}
 BUILD_DUCKDB="${BUILD_DUCKDB:-true}"
 export CMAKE_BUILD_TYPE=Release
 SUDO="${SUDO:-"sudo --preserve-env"}"
+USE_CLANG="${USE_CLANG:-false}"
+
+function install_clang15 {
+  VERSION=`cat /etc/os-release | grep VERSION_ID`
+  if [[ ! ${VERSION} =~ "22.04" && ! ${VERSION} =~ "24.04" ]]; then
```

Comment:
Have we had problems with clang on Ubuntu apart from these versions ? Maybe it would just be  good to add this as a warning rather than forcing the user to not use clang.

Reply:
The issue is that it is more complicated to get clang 15 onto the older Ubuntu 20.04. Perhaps we should just not support it anymore and only support the last two LTS versions? Because we officially say Ubuntu 20.04+ is supported I made this check - so really only affects users of Ubuntu 20.04.

Reply:
I still say it should be fine to spit out a warning ; we can assume folks know what they are doing if they choose to play around with compiler versions.